### DISCUSSION TITLE
All done. Your content was saved, but no Obsidian note was created

### DISCUSSION THREAD

**Discussion by crmfyi** [coddingtonbear/obsidian-web#172] [Bug Report]
### Describe the bug

In the past few weeks, I've been finding that when I use the clipper, I am not consistently getting content saved to Obsidian. I will be on a page, click the save button, get my success message, but when I go to Obsidian, the file has not been created. I go back to my browser and click the save button again and after I get the second success message, I then find that the note has been created. 

### To Reproduce

1. Go to a web page
2. Click on the Obsidian Web icon in the browser
3. Click the Send to Obsidian button
4. Go to Obsidian
5. Look for your clipped note
6. Go back to that web page again
7. Click on the Obsidian Web icon in the browser
8. Click the Send to Obsidian button
9. Go to Obsidian
10. Look for your clipped note - This time the note will be there

### Expected Behavior

In the past, my expectation of being able to rely on the success message truly indicating that the note has been saved was met. Today, I seem to have to do multiple saves and check each time to ensure I don't lose what I wanted to save

### Operating System

Mac OS Sonoma 14.13.1

### Browser

Arc

### Browser Version

Version 1.31.2 (46819) Chromium Engine Version 122.0.6261.57

**Comment by coddingtonbear** [comment]
I think I'd need to see the logs for this -- without that it's really hard for me to make any guesses since I do not personally experience this issue.

I'd have to guess, though, that this is instead going to be a situation in which the template you're using isn't doing quite what you're expecting it to do, or the changes made by Obsidian Web being overwritten by a misconfigured sync service (I know Obsidian's Sync service works, but I can't speak for using dropbox or anything else).

**Reply by coddingtonbear** [reply]
& just to make sure it's clear -- you can create the export from the extension options page; there's a bug-shaped icon in the upper-right corner.  What folks would need for troubleshooting is that you reproduce the problem (that is: you see the 'All done' message, but do not see the content in Obsidian), then immediately (being quick is important) go to the bug report screen on the options page, enable the "log messages" checkbox,  and then download/attach the export.

**Reply by crmfyi** [reply]
Here is the log file. Just before exporting the log, I tried saving a Cult of Mac article to Obsidian. I hope it helps you figure it out.
[Obsidian Bug Report (1).json](https://github.com/coddingtonbear/obsidian-web/files/14440810/Obsidian.Bug.Report.1.json)

**Reply by coddingtonbear** [reply]
I don't see any requests in the log you've uploaded that appear to be trying to modify your vault.  Are you sure you experienced the issue in the few minutes before you exported the log?

**Reply by crmfyi** [reply]
OK, I decided to wait overnight, pull up a page again, try to save it, look in Obsidian, didn't find it, so I went back to the browser, used the Obsidian Web extension again, and then exported the file. So what you should see is two attempts to use the extension to save to Obsidian. Only the second one actually saved. The created timestamp from Obsidian for that second save is 2024-02-29 09:05 AM. Thank you for your help looking into this. 
[Obsidian Bug Report (2).json](https://github.com/coddingtonbear/obsidian-web/files/14449809/Obsidian.Bug.Report.2.json)

**Reply by Tback1** [reply]
I have the same question, in #167 I notice I have to save twice too

**Reply by Tback1** [reply]
@coddingtonbear 
## Log:
[obsidian-local-rest-api.bug_report.json](https://github.com/coddingtonbear/obsidian-web/files/14461426/obsidian-local-rest-api.bug_report.json)

`如何维护职场权益` , please check this title

![image](https://github.com/coddingtonbear/obsidian-web/assets/8872546/7a03240d-7867-42e4-b55b-747ce3daed4b)

**Reply by coddingtonbear** [reply]
~~Looking at the path for the file we're trying to create @Tback1, it _does_ look like we're trying to create a file in your vault named just `.md` as you were describing in #176.  I'm afraid that that is almost certainly the fault of the template you are using for generating your snapshot.  If the title of your page includes no filename-safe characters, you will have to define a filename of your own manually, or use a different method for naming your page.~~

~~If you're able to confirm that Obsidian allows use of mandarin characters in created filenames, maybe you can post a pull request updating this handlebars helper such that it'll allow you to create your file: https://github.com/coddingtonbear/obsidian-web/blob/master/src/handlebars.ts#L28.~~

Actually, I think you're right -- there does appear to be some kind of problem around how we fetch or render page titles, maybe.

**Reply by coddingtonbear** [reply]
Interestingly, in _your_ output @crmfyi I do see that the same URL appears to have two different requests associated:

```json
    {
      "date": "2024-02-29T15:05:01.696Z",
      "level": "log",
      "message": "Incoming obsidian-request background request",
      "data": {
        "sender": {
          "id": "edoacekkjanmingkbkgjndndibhkegad",
          "url": "https://www.macrumors.com/2024/02/28/apple-refurbished-m3-imac/",
          "origin": "https://www.macrumors.com",
          "frameId": 0,
          "documentId": "DDFA6A657510892F2D955319DBC52C0A",
          "documentLifecycle": "active",
          "tab": {
            "active": true,
            "audible": false,
            "autoDiscardable": true,
            "discarded": false,
            "favIconUrl": "https://images.macrumors.com/images-new/favicon-32x32.png",
            "groupId": -1,
            "height": 1003,
            "highlighted": true,
            "id": 339100998,
            "incognito": false,
            "index": 13,
            "lastAccessed": 1709219086912.786,
            "mutedInfo": {
              "muted": false
            },
            "pinned": false,
            "selected": true,
            "status": "complete",
            "title": "Apple Begins Selling Refurbished iMac With M3 Chip - MacRumors",
            "url": "https://www.macrumors.com/2024/02/28/apple-refurbished-m3-imac/",
            "width": 1592,
            "windowId": 339100685
          }
        },
        "message": {
          "type": "obsidian-request",
          "request": {
            "path": "/vault/Spaces/Work/Projects/Effectv/Effectv%20Phase%201/Resources/.md",
            "options": {
              "method": "put",
              "body": "---\nType: Resource\nCompany: \"[[Effectv]]\"\nProject: \"[[Effectv Phase 1 Project Overview|Effectv Phase 1 Project]]\"\nAuthor: \ntags:\n - Effectv/Phase1\nSource: \nLink: https://www.macrumors.com/2024/02/28/apple-refurbished-m3-imac/\nSave Date: 2024-02-29 09:04:58\nup: [[Effectv Phase 1 Project Overview|Effectv Phase 1 Project]]\n---\nup: `= this.Up`\n# Apple Begins Selling Refurbished iMac With M3 Chip - MacRumors\nGo: `= this.Link`\n> [!SUMMARY] Summary\n> \n\n\n## Preview\n\n### Meetings\n```dataview\nTABLE WITHOUT ID\nfile.link AS Meeting, Date\nFrom [[]]\nWhere Type = \"Meeting\"\nSort Date DESC\n```\n",
              "headers": {
                "Content-Type": "text/markdown"
              }
            }
          }
        }
      },
      "stack": null
    }
```

and

```json
   {
      "date": "2024-02-29T15:05:20.595Z",
      "level": "log",
      "message": "Incoming obsidian-request background request",
      "data": {
        "sender": {
          "id": "edoacekkjanmingkbkgjndndibhkegad",
          "url": "https://www.macrumors.com/2024/02/28/apple-refurbished-m3-imac/",
          "origin": "https://www.macrumors.com",
          "frameId": 0,
          "documentId": "DDFA6A657510892F2D955319DBC52C0A",
          "documentLifecycle": "active",
          "tab": {
            "active": true,
            "audible": false,
            "autoDiscardable": true,
            "discarded": false,
            "favIconUrl": "https://images.macrumors.com/images-new/favicon-32x32.png",
            "groupId": -1,
            "height": 1003,
            "highlighted": true,
            "id": 339100998,
            "incognito": false,
            "index": 13,
            "lastAccessed": 1709219116538.662,
            "mutedInfo": {
              "muted": false
            },
            "pinned": false,
            "selected": true,
            "status": "complete",
            "title": "Apple Begins Selling Refurbished iMac With M3 Chip - MacRumors",
            "url": "https://www.macrumors.com/2024/02/28/apple-refurbished-m3-imac/",
            "width": 1592,
            "windowId": 339100685
          }
        },
        "message": {
          "type": "obsidian-request",
          "request": {
            "path": "/vault/Spaces/Work/Projects/Effectv/Effectv%20Phase%201/Resources/Apple Begins Selling Refurbished iMac With M3 Chip - MacRumors.md",
            "options": {
              "method": "put",
              "body": "---\nType: Resource\nCompany: \"[[Effectv]]\"\nProject: \"[[Effectv Phase 1 Project Overview|Effectv Phase 1 Project]]\"\nAuthor: \ntags:\n - Effectv/Phase1\nSource: \nLink: https://www.macrumors.com/2024/02/28/apple-refurbished-m3-imac/\nSave Date: 2024-02-29 09:05:18\nup: [[Effectv Phase 1 Project Overview|Effectv Phase 1 Project]]\n---\nup: `= this.Up`\n# Apple Begins Selling Refurbished iMac With M3 Chip - MacRumors\nGo: `= this.Link`\n> [!SUMMARY] Summary\n> \n\n\n## Preview\n\n### Meetings\n```dataview\nTABLE WITHOUT ID\nfile.link AS Meeting, Date\nFrom [[]]\nWhere Type = \"Meeting\"\nSort Date DESC\n```\n",
              "headers": {
                "Content-Type": "text/markdown"
              }
            }
          }
        }
      },
      "stack": null
    }
```

And those do confirm the lead that @Tback1 mentioned earlier -- clearly in one of your requests, we're attempting to store a file named just `.md` (and succeeding at doing so), and then in the subsequent one, we're actually setting a proper filename for it.

I guess I'm no longer convinced this is a regex problem as I'd suggested a moment ago to @Tback1; I even can clearly see that there is a title set by the time the request comes in.  This might be either a situation in which the title is changing (and just happens to be empty when the dialog is opened) or it could be a situation in which something is going wrong when we attempt to render the title template.

**Reply by coddingtonbear** [reply]
Could one of the two of you create a screencast of you reproducing this?  So far, I haven't been able to reproduce this problem locally, and looking at the code for this it looks pretty bulletproof that we'd gather the page context (which includes the page title) when the Obsidian Web popup's form is displayed (https://github.com/coddingtonbear/obsidian-web/blob/d840769ec5608119414c7b57c2b23d41b7d61646/src/popup.tsx#L406) and I wonder if maybe there's something about how you both are using this tool that I'm not expecting.

**Reply by Tback1** [reply]
I guess this maybe made also connected with the #167 which the page need refresh to show it's duped.
The save 100% could save right in the second ,and in the first it's wrong for name

### I  made the screencast:
1,2 link it had opened(or Cached?) so it's OK, the third link is first open,so the name just `.md`

![screenshots](https://github.com/coddingtonbear/obsidian-web/assets/8872546/aa4951ec-159d-4bdf-a6c9-f5632541414d)

**Reply by Tback1** [reply]
Let me conclude that there are several related phenomena to this problem：
1. If you do not refresh after saving, you will not be able to identify page duplicates
2. The first time the page was opened, the save file name was incorrect as `.md`.(by the way, the `.md` have the save content,just wrong name)
3. It needs to be saved twice, the first time there is problem 2, but the second time it must be fine

**Reply by coddingtonbear** [reply]
Just to be clear (1) is the intended behavior, as we discussed earlier.  We do not right now attempt to search for duplicates until you land on the page again in the future, and I'm not convinced changing that behavior is desirable given that I can't imagine how you'd not know that there was a duplicate if you just created it.

**Reply by Tback1** [reply]
First of all, in the version before the **badge err bug** #157 was fixed, this behavior was not like this, and it was allowed to save immediately after saving and then immediately to find duplicates. 

1.opening the save window immediately after saving to find duplicates will make me sure that the project was saved successfully. In other words, **the act of saving again immediately is not to check for duplicates, but to ensure that the save is successful**

2.I suspect that the code change behind this intended behavior is causing the problems that follow（2）（3）.

**Reply by Tback1** [reply]
And I note that there have been at least 4 discussions on this issue
#167 
#172
#175 
#176

During the recording [screencast](https://github.com/coddingtonbear/obsidian-web/discussions/172#discussioncomment-8697851), I mentioned earlier that it could be [a special character caused by an error in the save name](https://github.com/coddingtonbear/obsidian-web/discussions/176#discussion-6326778), which is **wrong direction**.

**Reply by coddingtonbear** [reply]
Regarding the badge error bug (#157): I'm afraid I'm basically certain this isn't true:

> First of all, in the version before the badge err bug https://github.com/coddingtonbear/obsidian-web/discussions/157 was fixed, this behavior was not like this, and it was allowed to save immediately after saving and then immediately to find duplicates.

In that version (and this one) the main mechanism we use for determining when we should check for duplicates (for the purposes of updating the badge, or, in the more recent version: automatically opening the dialog) is that your URL has changed (see https://github.com/coddingtonbear/obsidian-web/blob/master/src/service_worker.ts#L52).  Now we do check in a few other situations, but the change is in the direction of checking in _more_ situations than we were before rather than fewer.  But, if you're interested in proving that wrong, you are welcome to check out the version of Obsidian Web before that change to give it a shot.

Regarding the second point: I'm not sure how to parse "the act of saving again immediately is not to check for duplicates, but to ensure that the save is successful"; could you explain what you mean in a different way?  I'm not sure I understand what saving multiple times might have to do with checking for duplicates, and, if anything, would expect that saving multiple times would save your document multiple times instead.  But in any case, saving the document multiple times has no connection with finding duplicates that I can understand.

**Reply by coddingtonbear** [reply]
And, @Tback1 -- I understand that you feel like there's a connection between all of those issues, but in at least half of the cases you've presented the issue described is clearly and obviously unrelated.

One is describing that a search isn't successfully finding a match (this has nothing, as far as I can imagine, to do with a the 'Save' button not getting the document into your notes, and furthermore I still have not found a way of reproducing it: #167), and another is a misunderstanding of what a particular operation is intended to do (#175) -- they're complaining that an operation that's intended to append content every time it's invoked is... doing exactly that: appending content every time it's invoked.

**Reply by Tback1** [reply]
I'm not saying that I really want to save twice, but in the previous version, I saved once and it was OK, but sometimes I would immediately use the shortcut key to call up the save window again, and if there were duplicates, I would know that the previous save was successful, instead of switching to Obsidian to see if the save was successful.

I checked and  copied the version in #157(obsidian-web 3.1,Local REST API 2.0.4,the Chrome I updated frome 118 to 120)
so in that environment, Error badge aside, it's okay to save, but after fixing patch #157, there was a save failure problem (and no duplicates found even if the refresh was refreshed), and the fix at the time, [I remember making a patch to Local REST first](https://github.com/coddingtonbear/obsidian-web/discussions/157#discussioncomment-8199007), so I wonder if it's a problem over there?

**Or the Chrome updated to 120 make this? `chrome.storage.local` `chrome.storage.sync` behavier changed,so like the first opened page get title error?**

So anyway I don't know any more information on this question. There are so many puzzles and possibilities to scratch your head

**Reply by crmfyi** [reply]
Thanks for looking into this @coddingtonbear . I have tested this a few more times today, and I'm not running into the issue yet at all. It seems to be saving on the first request each time.  Knowing what you discovered about the Title token not being placed before the .md is a great catch for me to now expand the extension and just see if the data is captured in it. I suppose there could be something that was happening with a particular version of Arc and the Title as well, seeing as Arc releases a new version each week. I'll go in and try saving a few more times specifically and see what my success rate is, then get back to you.

**Reply by coddingtonbear** [reply]
I'm guessing things have worked out alright so far, @crmfyi & am going to mark this issue as resolved, but let me know if you did continue to encounter problems.

**Reply by crmfyi** [reply]
Everything is good right now. Thank you for jumping on this quickly

**Comment by coddingtonbear** [comment]
While looking through things, I found a little bug that seems like it might be the cause of this:

63d6609470e543efec49ecb8b214f37c7383be9e

this bug would've prevented the generated "API URL" from being re-computed as your preview context (that is: the page url, title, content) are changed.  This means that once your computed URL was set once, that it would never change even if the page title (or any other context) changed.  I could imagine how that might cause this problem if for some reason when the form was displayed at first, no page context was yet available (and thus, no API URL was set).  If you were then to close and re-open the form, that component would be unmounted and re-mounted, but in that later re-opening, we'd already have gathered page context, so its initial calculation _would_ have a value.

I'm going to put this and a few other related changes together into another release tonight in hopes that that'll wrap this problem up completely.

**Reply by coddingtonbear** [reply]
I've posted v3.3.1 for approval in the Chrome Web Store.  Usually they review in a few hours.  If you want to force-update your extension once it's released, see https://github.com/coddingtonbear/obsidian-web/wiki/How-do-I-ask-Chrome-to-update-extensions-immediately%3F.

**Reply by huangjingyun8** [reply]
I went back to test it tonight. This problem has been bothering me for a few days. Especially friends who use chrome and arc together may have experienced this confusion.

An irresponsible speculation of mine: the recent wave of problems may be caused by the frequent updates of arc.

**Reply by coddingtonbear** [reply]
I really don't know much about Arc or Brave or Opera or any of the other very-uncommon browsers out there and only help folks out when they're using Chrome, but I'd be kind of surprised if the problem in this case were rooted in something specific to Arc.  _Especially_ since I found that bug I noted above that looks like a plausible culprit, but I'm curious to hear if once folks get the updated version whether they're still experiencing this issue.
